### PR TITLE
Updates ouroboros-network to fix PeerSharing

### PR DESCRIPTION
**Pull request by bolt12**
# Description

This PR accommodates changes from https://github.com/input-output-hk/ouroboros-network/pull/4644.

### CODE REVIEW

File: ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md
Comment (nfrisby) [inline ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md:15]: - `PeerSharingDisabled` was renamed to `NoPeerSharing` upstream. I don't think that needs to be mentioned in the Consensus change log.
- On the other hand, I would imagine it does mean that the `build-depends` field should be updated with an greater lower bound?

File: ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md
Comment (nfrisby) [inline ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md:19]: `immDBServer` is not technically part of our interface, since it's in an "unstable-*" package. I therefore think this change should instead be at most a Patch-level item, or even just not mentioned in the changelog.

File: ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Network/NodeToNode.hs
Comment (nfrisby) [inline ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Network/NodeToNode.hs:286]: Looks like this is only called in the `ouroboros-consensus` repo's code in the `ouroboros-consensus-diffusion`'s `Ouroboros.Consensus.Node` module and the ThreadNet tests.

This diff alters the ThreadNet call, but it doesn't alter the `Ouroboros.Consensus.Node` call, since that one is using arguments from the Diffusion Layer packages whose type has recently changed, and which is what motivates the change here in the first place.

**Review by nfrisby — CHANGES_REQUESTED** [review]
My only requested changes are:

- I think a couple of the changelog fragments are incorrect, see my comments there.

- I suspect at least one `build-depends` lower bound needs to be increased.

File: ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md
Comment (bolt12) [reply ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md:19]: Got it, thanks!

File: ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Network/NodeToNode.hs
Comment (bolt12) [reply ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Network/NodeToNode.hs:286]: What are you trying to say?

File: ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md
Comment (bolt12) [reply ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md:15]: Yes, but should I do that before network patches get a release?

Comment (nfrisby) [reply ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md:15]: I think so, but I don't know if any such policy is technically established.

A related thought: the PR cannot merge until CI passes, and CI won't pass until such a network release is on CHaP.

Comment (bolt12) [reply ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md:15]: I think that is alright, the PR is small and can wait to be merged until everything is in place

**Comment by bolt12** [comment]
I believe I need to wait for `plutus` to make a release using most recent version of `nothunks`, before merging

**Review by jasagredo — APPROVED** [review]

File: ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md
Comment (jasagredo) [reply ouroboros-consensus-cardano/changelog.d/20231002_154149_armandoifsantos_fix_peerSharing_handshake.md]: Ouroboros-network is released, and I modified the changelog fragments. I will dismiss this request for changes.

**Pull request merged**
